src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -91,6 +113,33 @@ public function generateUniqueCode()
      */
     protected function isUsedCode($code)
     {
-        return null !== $this->repository->findOneBy(['code' => $code]);
+        foreach ($this->generatedCoupons as $coupon) {
+            if ($coupon->getCode() === $code) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param InstructionInterface $instruction
+     */
+    protected function initialize(InstructionInterface $instruction)
+    {
+        $this->isGenerationPossible($instruction);
```

Comment:
I'd think that this returns true/false. Should be ``assertGenerationIsPossible($instruction)``.